lib/ansible/modules/storage/netapp/na_ontap_volume.py

```diff
@@ -78,7 +78,9 @@
   type:
     description:
     - The volume type, either read-write (RW) or data-protection (DP).
-
+    choices: ['rw', 'dp']
```

Comment:
As @carchi8py pointed out, the ZAPI documentation shows 4 possible values:

- 'rw' - read-write volume (default),
- 'ls' - load-sharing volume,
- 'dp' - data-protection volume,
- 'dc' - data-cache volume (FlexCache)

(this is from SDK documentation for 9.5)
even though the CLI allows only 2 values.

I'll check with the product team why there is such an inconsistency.

This is also why I am always conflicted with choices.  On one hand, this is a great way to help the end users.  On the other hand, we need to support all possible choices for 9.1 to 9.6.  And if 9.7 introduces a new value, it will require a change in the module.  While a free format fields give a lot of flexibility.

Reply:
I'm starting to wonder if this is a bug in the ZAPI documentation.   There are only 2 places where ls and dc are mentioned.

We should try and see if the ZAPI actually accept these values.

Anyway, this is not a critical fix, so it's better to delay until we know for sure this change would not break anything.